packages/eslint-plugin/rules/__tests__/react-no-unsafe-timeout.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { RuleTester } from 'eslint';
+
+/**
+ * Internal dependencies
+ */
+import rule from '../react-no-unsafe-timeout';
+
+const ruleTester = new RuleTester( {
+	parserOptions: {
+		ecmaVersion: 6,
+	},
+} );
+
+ruleTester.run( 'react-no-unsafe-timeout', rule, {
+	valid: [
+		{
+			code: `function getNotComponent() { setTimeout(); }`,
+		},
+		{
+			code: `function MyComponent( props ) { const { setTimeout } = props; ( () => { setTimeout(); } )(); }`,
+		},
+		{
+			code: `function MyComponent( props ) { props.setTimeout(); }`,
+		},
+		{
+			code: `class MyNotComponent { doAction() { setTimeout(); } }`,
+		},
+		{
+			code: `class MyComponent extends wp.element.Component { componentDidMount() { const { setTimeout } = this.props; setTimeout(); } }`,
+		},
+		{
+			code: `class MyComponent extends Component { componentDidMount() { const { setTimeout } = this.props; setTimeout(); } }`,
+		},
+		{
+			code: `class MyComponent extends Component { componentDidMount() { this.props.setTimeout(); } }`,
+		},
+		{
+			code: `class MyComponent extends Component { componentDidMount() { this.timeoutId = setTimeout(); } }`,
```

Comment:
It still can be invalid (not canceled when the component gets unmount) but I think it's fine to assume good intents and keep implementation reasonable :)

Reply:
> It still can be invalid (not canceled when the component gets unmount) but I think it's fine to assume good intents and keep implementation reasonable :)

Right, the thinking was: You'd only ever care about the value produced by `setTimeout` if you intended to use it in `clearTimeout` (the return value is [described as such](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Return_value)). There's no guarantee here that this actually occurs, or that it occurs specifically in `componentWillUnmount`. But it's also the only option for considering "safe" in a generic sense where someone may want to use this rule, but wouldn't use `withSafeTimeout` from `@wordpress/compose`.